packages/gen-ai-hub/src/core/http-client.ts

```diff
@@ -38,73 +19,45 @@ export type CustomRequestConfig = Omit<
  */
 export interface EndpointOptions {
   /**
-   * The specific endpoint to call.
+   * The deployment ID to call.
    */
-  url: string;
-
+  deploymentId: string;
   /**
-   * The API version to use.
+   * The specific endpoint to call.
    */
-  apiVersion?: string;
+  path: string;
 }
+
 /**
  * Executes a request to the AI Core service.
  * @param endpointOptions - The options to call an endpoint.
  * @param data - The input parameters for the request.
  * @param requestConfig - The request configuration.
  * @returns The {@link HttpResponse} from the AI Core service.
  */
-export async function executeRequest<D extends BaseLlmParameters>(
+export async function executeRequest(
   endpointOptions: EndpointOptions,
-  data: D,
+  data: any,
   requestConfig?: CustomRequestConfig
 ): Promise<HttpResponse> {
   const aiCoreDestination = await getAiCoreDestination();
-  const { deploymentConfiguration, ...body } = data;
-  const { url, apiVersion } = endpointOptions;
-
-  const mergedRequestConfig = {
-    ...mergeWithDefaultRequestConfig(apiVersion, requestConfig),
-    data: JSON.stringify(body)
-  };
 
   const targetUrl =
     aiCoreDestination.url +
     '/v2/inference/deployments/' +
```

Comment:
> [q] Can we not pass /v2/inference/deployments/ part of the url also from the openai/orchestration chatCompletion method?

we could, but to what benefit? The assumption of this method is that it will always serve inference requests, otherwise the `EndpointOptions` doesn't make sense anymore.

We could consider having two methods, something like: `executeRequest(path, payload)` and `executeInferenceRequest(EndpointOptions, payload)`.

> For the apiVersion, the check params: apiVersion ? { 'api-version': apiVersion } : {} should ensure that its passed only when needed? EndpointOptions is internal anyway, so we have control over when this needs to be passed?

Here IDK what you mean..

Reply:
This was not originally intended to serve only inference requests, atleast at the time we considered only llm access requests and not ai-core at all. Hence hardcoded the inference part. This was meant as a wrapper around the generic http-client from Cloud SDK for all requests (unless the req are completely different). In that direction, my thought was to take the `/v2/inference/deployments/` part away from this function and pass it instead from the respective function as `/v2/inference/deployments/completion`. Then the `path` is merged to the `baseUrl` fetched from the destination to create the final url.

The request config merge code feel duplicated to me atm with the exception of `apiVersion`. This is optional in `EndpointOptions` anyway and is not set by the user. I assumed with this check `params: apiVersion ? { 'api-version': apiVersion } : {}` we can ensure this is set only when needed. 

Anyway, I didn't review the whole PR so I might be missing context. If there are reasons why this function cannot stay generic, then its fine.

Reply:
Ah okay, got it. Yes, I think we can make this generic, there is no particular reason for the current design. In this PR, I just went with what I felt makes adding the deployment ID lookup and testing easier, but this wasn't the main focus of this PR of course.

I would suggest we extract this entire change out of this PR and do it as a follow up to #61 / with the fix to the execute method.

I'll probably anyway have to reset this PR later on, removing anything that is not explicitly the constants + deployment ID lookup 😄